src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexWriter.cs

```diff
@@ -80,7 +82,12 @@ public void Commit(IState state, IndexingStatsScope commitStats)
             finally
             {
                 _indexWriter.SetCommitStats(null);
-                RecreateIndexWriter(state);
+
+                if (RecreateAfterCommit)
+                {
+                    RecreateAfterCommit = false;
```

Comment:
In addition to `LuceneIndexWriter` we also have `LuceneSuggestionIndexWriter` where also `RecreateIndexWriter()` is called
 
 https://github.com/ravendb/ravendb/blob/a868f1422b7edc40064728e4d583585209cf899f/src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneSuggestionIndexWriter.cs#L204
 
 So I believe it needs to follow the same logic as here